lingpy/basictypes.py

```diff
@@ -0,0 +1,52 @@
+from array import array
+
+
+class strings(list):
+
+    def __init__(self, iterable):
+        
+        list.__init__(self, iterable if not isinstance(iterable, str) else
+                iterable.split())
+
+    def __str__(self):
+
+        return ' '.join(self)
+
+
+class ints(list):
+
+    def __init__(self, iterable):
+
+        list.__init__(self, [int(x) for x in (iterable if
+           not isinstance(iterable, str) else iterable.split())])
+
+    def __str__(self):
+
+        return ' '.join([str(x) for x in self])
+
+
+class floats(list):
```

Comment:
actually, let's try this:
```python
from array import array
from functools import partial

class _array(array):
    def __str__(self):
        return ' '.join([str(x) for x in self])

ints = partial(_array, 'i')
floats = partial(_array, 'f')
```

Reply:
Problem with `array` is here: https://bytes.com/topic/python/answers/484386-inherit-array

I get strange init errors when using this in py3.

Reply:
```
from array import array
from functools import partial

class _array(array):

    def __init__(self, type_, iterable):
        
        array.__init__(self, type_, iterable)

    def __str__(self):
        return ' '.join([str(x) for x in self])

ints = partial(_array, 'i')
floats = partial(_array, 'f')

```

the error is

```
In [8]: _array('i', [1, 2, 34])
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-8-ca395f82e363> in <module>()
----> 1 _array('i', [1, 2, 34])

/home/jmlist/projects/scripts/LingPy/dev/lingpy/lingpy/basictypes.py in __init__(self, type_, iterable)
      6     def __init__(self, type_, iterable):
      7 
----> 8         array.__init__(self, type_, iterable)
      9 
     10     def __str__(self):

TypeError: object.__init__() takes no parameters

```